### PR TITLE
build: do not produce Windows binaries with netgo build tags

### PR DESCRIPTION
**Pull request by rfratto**
Go 1.19 adds Windows support for the native Go network stack, but doesn't include support for resolving DNS short names.

Other projects, including Prometheus, have updated their build process to exclude the netgo build tags when producing Windows binaries to work around this behavior.

Fixes grafana/agent#4665

### CODE REVIEW

**Comment by ptodev** [comment]
I think you mean that it fixes #4665 and not #4465 ?

**Comment by rfratto** [comment]
> I think you mean that it fixes https://github.com/grafana/agent/issues/4665 and not https://github.com/grafana/agent/issues/4465 ?

🤦 Yes, thank you

File: tools/make/packaging.mk [hunk 1]
```diff
@@ -31,63 +31,65 @@ dist-agent-binaries: dist/grafana-agent-linux-amd64       \
...
 
-dist/grafana-agent-darwin-arm64: GO_TAGS += builtinassets
+dist/grafana-agent-darwin-arm64: GO_TAGS += netgo builtinassets
 dist/grafana-agent-darwin-arm64: GOOS    := darwin
 dist/grafana-agent-darwin-arm64: GOARCH  := arm64
 dist/grafana-agent-darwin-arm64: generate-ui
 	$(PACKAGING_VARS) AGENT_BINARY=$@ $(MAKE) -f $(PARENT_MAKEFILE) agent
 
+# NOTE(rfratto): do not use netgo when building Windows binaries, which
+# prevents DNS short names from being resovable. See grafana/agent#4665.
```
Comment (ptodev) [inline tools/make/packaging.mk:71]: Do we need to add a TODO to revert this change when the underlying Go issue is resolved?
https://github.com/golang/go/issues/57757

Comment (rfratto) [reply tools/make/packaging.mk:71]: I don't think that's the same issue, is it? I don't see any mention of the hosts file in the issue or the Prometheus issue.

Comment (ptodev) [reply tools/make/packaging.mk:71]: Ah yes, I think you're right and it's not the same issue. But it's probably another issue which needs to be resolved before we use netgo on windows. Anyway, thank you for adding the TODO.
